Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -193,13 +222,61 @@ public static Song ReadMidi(string path, ref CallbackState callBackState)
                         {
                             instrument = Song.Instrument.Unrecognised;
                         }
+                        else if (song.ChartExistsForInstrument(instrument))
+                        {
+                            messageList.Add(new MessageProcessParams()
+                            {
+                                message = $"A track was already loaded for instrument {instrument}, but another track was found for this instrument: {trackNameKey}\nWould you like to overwrite the currently loaded track?",
+                                title = "Duplicate Instrument Track Found",
+                                executeInEditor = false,
+                                currentSong = song,
+                                trackNumber = i,
+                                processFn = (MessageProcessParams processParams) => {
+                                    Debug.Log($"Overwriting already-loaded part {processParams.instrument}");
+                                    foreach (Song.Difficulty difficulty in EnumX<Song.Difficulty>.Values)
+                                    {
+                                        var chart = processParams.currentSong.GetChart(processParams.instrument, difficulty);
+                                        chart.Clear();
+                                        chart.UpdateCache();
+                                    }
+
+                                    ReadNotes(midi.Events[processParams.trackNumber], processParams.currentSong, processParams.instrument);
+                                }
+                            });
+                            break;
+                        }
```

Comment:
Since we're in a switch case here, the `break` after the message addition skips the ReadNotes call already. Would probably be clearer to add a final `else` branch though.

Reply:
Oh, I see it now. That doesn't read well, if you're in a switch statement the break statement should always be expected to be at the end, that defs needs changing. Relying on break statements to control flow isn't much better than using gotos. There are some cases where it makes sense, but should be minimised as much as possible.

Reply:
Alrighty, got that clarified.